### PR TITLE
Improve Node, Taints, and Tolerations concepts

### LINKED ISSUE #17846: Issue with k8s.io/docs/tasks/administer-cluster/safely-drain-node/
This is a Bug Report

Problem:
Nowhere in the whole document the node status is mentioned as "Ready,SchedulingDisabled". 

Proposed Solution:
Update both https://kubernetes.io/docs/concepts/architecture/nodes/#manual-node-administration and k8s.io/docs/tasks/administer-cluster/safely-drain-node/ to describe correct node status

Page to Update:
https://kubernetes.io/docs/concepts/architecture/nodes/#manual-node-administration
k8s.io/docs/tasks/administer-cluster/safely-drain-node/ to describe correct node status

### PR DESCRIPTION
**Pull request by sftim**
- Reword https://kubernetes.io/docs/concepts/architecture/nodes/ ([preview](https://deploy-preview-17509--kubernetes-io-master-staging.netlify.com/docs/concepts/architecture/nodes/))
- Reword https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/ ([preview](https://deploy-preview-17509--kubernetes-io-master-staging.netlify.com/docs/concepts/configuration/taint-and-toleration/))

See [list of commits](https://github.com/kubernetes/website/pull/17509/commits) for details.

Now also fixes #17846.

### CODE REVIEW

File: content/en/docs/concepts/architecture/nodes.md [hunk 1]
```diff
@@ -237,17 +237,16 @@ checks the state of each node every `--node-monitor-period` seconds.
...
-when there is some change or enough time has passed (default is 1 minute, which
-is longer than the default timeout of 40 seconds for unreachable nodes). Since
-node lease is much more lightweight than NodeStatus, this feature makes node
-heartbeat significantly cheaper from both scalability and performance
-perspectives.
-
-In most cases, node controller limits the eviction rate to
+
+When NodeLease is enabled, each Node has an associated Lease object in the
+`kube-node-lease` namespace that the kubelet periodically renews. The control
```
Comment (kbhawkey) [inline content/en/docs/concepts/architecture/nodes.md]: @sftim, how is this change related to PR #17189?

**Comment by tengqm** [comment]
/lgtm

I have a concern regarding this kind of changes. It reads to me like a reorganization of  the existing contents which may make the content flows more smoothly. Since the content is now translated into several languages, this kind of changes will trigger a rework in all non-English repos. Maybe we want to be a little bit more refrained from doing this.

File: content/en/docs/concepts/architecture/nodes.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/concepts/architecture/nodes.md]: PR #17189 is about promoting NodeLease to GA. Both cover the same text and if this one lands first I was expecting to help out with tweaks / a rebase for #17189

File: content/en/docs/concepts/architecture/nodes.md [hunk 2]
```diff
@@ -81,25 +81,26 @@ The node condition is represented as a JSON object. For example, the following r
...
+have stopped running in your cluster. You can see the pods that might be running on
+an unreachable node as being in the `Terminating` or `Unknown` state. In cases where
+Kubernetes cannot deduce from the underlying infrastructure if a node has permanently
+left a cluster, the cluster administrator may need to delete the node object by hand.
+Deleting the node object from Kubernetes causes all the Pod objects running on the
+node to be deleted from the apiserver, and frees up their names.
+
+If you have the `TaintNodesByCondition`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled
+(the default), then the node lifecycle controller automatically creates
```
Comment (kbhawkey) [inline content/en/docs/concepts/architecture/nodes.md]: nit: If the TaintNodesByCondition is enabled by default, do you need lines 92- beginning of 94? 
Same comment could be applied to lines  231-233 and lines 285-286.

File: content/en/docs/concepts/architecture/nodes.md [hunk 3]
```diff
@@ -282,16 +281,19 @@ that the scheduler won't place Pods onto unhealthy nodes.
 
 ### Node capacity
 
-The capacity of the node (number of cpus and amount of memory) is part of the node object.
-Normally, nodes register themselves and report their capacity when creating the node object. If
-you are doing [manual node administration](#manual-node-administration), then you need to set node
-capacity when adding a node.
+Node objects include the Node's resource capacity (for example: the amount of memory
+available, and the number of CPUs).
```
Comment (kbhawkey) [inline content/en/docs/concepts/architecture/nodes.md]: - This page gets confusing when referring to the Node (object/resource) versus the node (physical machine, virtual machine). I _think_ it is acceptable to refer to the Node (object) as just Node (capitalized).
For instance, see line 116. NodeStatus is a field in the Node object (not the node). Line 12 starts with _Node_ is a worker *machine*. Should this be: a _Node_ represents a physical or virtual machine? Here the sentence is referring to the object not the machine.
- Also check node controller vs NodeController, Node Controller
- Line 284: Nodes maintain information about a node's resource capacity (for example: ...).
- Check lines 286 - 289. I am having trouble parsing this sentence.

File: content/en/docs/concepts/architecture/nodes.md (hunk 1 above)
Comment (kbhawkey) [reply content/en/docs/concepts/architecture/nodes.md]: Yes, that is true, however, the GA related content is nearly identical and is intended for the upcoming 1.17 release.
This change will definitely interfere with that merge.

File: content/en/docs/concepts/architecture/nodes.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/concepts/architecture/nodes.md]: That  sounds like a good idea for an improvement, maybe for a future PR.

File: content/en/docs/concepts/architecture/nodes.md (hunk 3 above)
Comment (sftim) [reply content/en/docs/concepts/architecture/nodes.md]: Thanks for the feedback.

> Line 12 starts with Node is a worker machine. Should this be: a Node represents a physical or virtual machine? Here the sentence is referring to the object not the machine.

I'm OK with the idea that _all_ objects in computing are in some sense an abstraction&mdash;same way that PersistentVolume or Machine (from the cluster API) are also abstract. For that reason I prefer either the existing wording or my proposed rewrite.

Where a sentence starts with “Node&hellip;”, I think it helps to clarify “Node object” because at the start of a sentence Node will be capitalized whatever it refers to/

File: content/en/docs/concepts/architecture/nodes.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/concepts/architecture/nodes.md]: When features are beta, I think it helps to let readers know how to opt out of them.

File: content/en/docs/concepts/architecture/nodes.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/concepts/architecture/nodes.md]: Maybe the easy option is to leave this until v1.17 is out?

**Comment by sftim** [comment]
> Since the content is now translated into several languages, this kind of changes will trigger a rework in all non-English repos. Maybe we want to be a little bit more refrained from doing this.

Hmm, good point. I'm not sure what the right approach is here. A “pull” model would limit improvements to English documentation so that all localizations could catch up.

**Comment by sftim** [comment]
/retitle [WIP] Improve Node, Taints, and Tolerations concepts

**Comment by sftim** [comment]
/retitle Improve Node, Taints, and Tolerations concepts

File: content/en/docs/reference/glossary/node.md [hunk 4]
```diff
@@ -15,3 +15,5 @@ tags:
 <!--more--> 
 
 A worker node may be a VM or physical machine, depending on the cluster. It has local daemons or services necessary to run {{< glossary_tooltip text="Pods" term_id="pod" >}} and is managed by the control plane. The daemons on a node include {{< glossary_tooltip text="kubelet" term_id="kubelet" >}}, {{< glossary_tooltip text="kube-proxy" term_id="kube-proxy" >}}, and a container runtime implementing the {{< glossary_tooltip text="CRI" term_id="cri" >}} such as {{< glossary_tooltip term_id="docker" >}}.
+
+In early Kubernetes versions, Nodes were called “Minions”.
```
Comment (kbhawkey) [inline content/en/docs/reference/glossary/node.md]: nit: lowercase minions or remove this sentence?

File: content/en/docs/concepts/architecture/nodes.md [hunk 5]
```diff
@@ -9,11 +9,10 @@ weight: 10
 
 {{% capture overview %}}
 
-A _Node_ is a worker machine in Kubernetes (previously known as a `minion`). A node
-may be a virtual or physical machine, depending on the cluster. Each node contains
-the services necessary to run {{< glossary_tooltip text="Pods" term_id="pod" >}},
-managed by the {{< glossary_tooltip text="control plane" term_id="control-plane" >}}
-components.
+A _Node_ is a worker machine in Kubernetes. A node may be a virtual or physical
+machine, depending on the cluster. Each node contains the services necessary to
```
Comment (kbhawkey) [inline content/en/docs/concepts/architecture/nodes.md]: nit: Is the reference to `worker machine` essential?
Instead:
A Node [the object?] represents a virtual or physical machine, ...
A node is a virtual or physical machine ...

File: content/en/docs/concepts/architecture/nodes.md [hunk 6]
```diff
@@ -111,13 +111,14 @@ This information is gathered by Kubelet from the node.
...
 ## Management
 
-Unlike [pods](/docs/concepts/workloads/pods/pod/) and [services](/docs/concepts/services-networking/service/),
-a node is not inherently created by Kubernetes: it is created externally by cloud
-providers like Google Compute Engine, or it exists in your pool of physical or virtual
-machines. So when Kubernetes creates a node, it creates
-an object that represents the node. After creation, Kubernetes
-checks whether the node is valid or not. For example, if you try to create
-a node from the following content:
+There are two main ways to have Nodes added to the {{< glossary_tooltip text="API server" term_id="kube-apiserver" >}}:
```
Comment (kbhawkey) [inline content/en/docs/concepts/architecture/nodes.md]: nit (possible edit): There are two ways to create a node with the Kubernetes API server:
1. The kubelet registers ...
1. Configure a Node object and send this object as a request to the API server [kubectl, curl]
 
nit: I think this part is missing from your addition but could be reworded:
`When Kubernetes creates a node, it creates
an object that represents the node.`

File: content/en/docs/concepts/configuration/taint-and-toleration.md [hunk 7]
```diff
@@ -294,3 +294,10 @@ tolerations to all daemons, to prevent DaemonSets from breaking.
 
 Adding these tolerations ensures backward compatibility. You can also add
 arbitrary tolerations to DaemonSets.
+
+{{% /capture %}}
+{{% capture whatsnext %}}
+* Read about [out of resource handling](/docs/tasks/administer-cluster/out-of-resource/) and how you can configure it
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/taint-and-toleration.md]: nit: line 300, Read about managing node resources?

File: content/en/docs/concepts/architecture/nodes.md [hunk 8]
```diff
@@ -300,15 +298,15 @@ If you want to explicitly reserve resources for non-Pod processes, follow this t
...
-[Node API object](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#node-v1-core).
+See [Control Topology Management Policies on a Node](/docs/tasks/administer-cluster/topology-manager/)
+for more information.
 
 {{% /capture %}}
 {{% capture whatsnext %}}
-* Read about [node components](/docs/concepts/overview/components/#node-components)
-* Read about node-level topology: [Control Topology Management Policies on a node](/docs/tasks/administer-cluster/topology-manager/)
+* Learn about the [components](/docs/concepts/overview/components/#node-components) that make up a node.
+* Read the [API definition for Node](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#node-v1-core)
```
Comment (kbhawkey) [inline content/en/docs/concepts/architecture/nodes.md]: nit: No period at end of list items?

File: content/en/docs/concepts/configuration/taint-and-toleration.md [hunk 9]
```diff
@@ -71,7 +71,7 @@ A toleration "matches" a taint if the keys are the same and the effects are the
 * the `operator` is `Exists` (in which case no `value` should be specified), or
 * the `operator` is `Equal` and the `value`s are equal
 
-`Operator` defaults to `Equal` if not specified.
+`operator` defaults to `Equal` if not specified.
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/taint-and-toleration.md]: @sftim , This section is confusing.
This is a new sentence (capitalized previously)?
Instead of starting with `operator`, could modify to:
The `operator` defaults to ... OR
If not specified, `operator` defaults to `Equal`.
What do you think?

**Comment by kbhawkey** [comment]
@sftim , I have not read through every change in detail. I've noted several nits and questions.
The changes contain some cleanup and rewording of concepts.
At this point, I would recommend squashing the commits because a few changes in later commits override changes in previous commits. Would it be helpful to get another technical `lgtm`,
/cc @kubernetes/sig-node-pr-reviews

**Comment by sftim** [comment]
/retitle [WIP] Improve Node, Taints, and Tolerations concepts

**Comment by zacharysarah** [comment]
@sftim 👋 Just a heads up that this needs a rebase. I see this is a long running issue with recent activity (~14 days), so I assume you're still working on it.

File: content/en/docs/reference/glossary/node.md (hunk 4 above)
Comment (sftim) [reply content/en/docs/reference/glossary/node.md:19]: Nodes _were_ called Minions, with a capital M. I think this sentence is correct.

File: content/en/docs/concepts/architecture/nodes.md (hunk 5 above)
Comment (sftim) [reply content/en/docs/concepts/architecture/nodes.md]: I've reworded - how's this?

File: content/en/docs/concepts/architecture/nodes.md (hunk 6 above)
Comment (sftim) [reply content/en/docs/concepts/architecture/nodes.md]: I reworded this.

File: content/en/docs/concepts/configuration/taint-and-toleration.md (hunk 9 above)
Comment (sftim) [reply content/en/docs/concepts/configuration/taint-and-toleration.md]: Reworded

**Comment by sftim** [comment]
How's this? Once people are happy with the new content I'll squash and let it get /lgtm and /approve

**Comment by sftim** [comment]
/retitle Improve Node, Taints, and Tolerations concepts

**Comment by jimangel** [comment]
/lgtm
@sftim do you think this needs a tech `lgtm` too?

File: content/en/docs/concepts/architecture/nodes.md (hunk 8 above)
Comment (sftim) [reply content/en/docs/concepts/architecture/nodes.md:338]: The fix is to add a period after each item?

**Comment by sftim** [comment]
> @sftim do you think this needs a tech lgtm too?

@jimangel it's mostly changing the order of existing text. My gut feeling is it's fine without tech review.

**Comment by sftim** [comment]
Squashed commits

File: content/en/docs/concepts/architecture/nodes.md (hunk 8 above)
Comment (kbhawkey) [reply content/en/docs/concepts/architecture/nodes.md:338]: This is a nit. The fix is to be consistent.

**Comment by kbhawkey** [comment]
Additional page previews:
https://deploy-preview-17509--kubernetes-io-master-staging.netlify.app/docs/concepts/overview/components/

https://deploy-preview-17509--kubernetes-io-master-staging.netlify.app/docs/reference/glossary/?fundamental=true#term-node

File: content/en/docs/concepts/configuration/taint-and-toleration.md [hunk 10]
```diff
@@ -213,9 +214,8 @@ running on the node as follows
  * pods that tolerate the taint with a specified `tolerationSeconds` remain
    bound for the specified amount of time
 
-In addition, Kubernetes 1.6 introduced alpha support for representing node
-problems. In other words, the node controller automatically taints a node when
-certain condition is true. The following taints are built in:
+The node controller automatically taints a Node when certain condition is true.
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/taint-and-toleration.md:217]: when a certain condition is true
OR
when certain conditions are true

File: content/en/docs/concepts/configuration/taint-and-toleration.md [hunk 11]
```diff
@@ -236,19 +236,18 @@ with `NoExecute` effect. If the fault condition returns to normal the kubelet or
...
-as the master becoming partitioned from the nodes.
+The control plane limits the rate of adding node new taints to nodes. This rate limiting
+manages the number of evictions that are triggered when many nodes become unreachable at
+once (for example: if there is a network disruption).
 {{< /note >}}
 
-The feature, in combination with `tolerationSeconds`, allows a pod
-to specify how long it should stay bound to a node that has one or both of these problems.
+You can specify `tolerationSeconds` for a Pod to define how long that Pod should stay bound
+to a Node that has problems.
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/taint-and-toleration.md:245]: @sftim, I am just noting content that could use future edits or cleanup.
possible edit:
You can specify `tolerationSeconds` for a Pod to define how long that Pod stays bound
to a failing [or unresponsive] Node.

**Comment by kbhawkey** [comment]
@sftim, I read through the pages again. There are some content areas of the taint and toleration page that are overly complicated (somewhat hard to follow). I realize that this pull request did not focus on that.
/lgtm

**Comment by sftim** [comment]
A few more fixes. Previous LGTM was in https://github.com/kubernetes/website/pull/17509#issuecomment-620324987
